app/scripts/utils/track-utils.js

```diff
@@ -0,0 +1,404 @@
+  // 1024 points per tile vs 256 for 2D tiles
+  if (tilesetInfo.resolutions) {
+    const zoomIndexX = calculateZoomLevelFromResolutions(
+      tilesetInfo.resolutions, this._xScale,
+      tilesetInfo.min_pos[0], this.tilesetInfo.max_pos[0] - 2
+    );
+
+    return zoomIndexX;
+  }
+
+  // the tileProxy calculateZoomLevel function only cares about the
+  // difference between the minimum and maximum position
+  const xZoomLevel = calculateZoomLevel(xScale,
+    tilesetInfo.min_pos[0],
+    tilesetInfo.max_pos[0],
+    tilesetInfo.bins_per_dimension || tilesetInfo.tile_size);
+
+  let zoomLevel = Math.min(xZoomLevel, maxZoom);
+  zoomLevel = Math.max(zoomLevel, 0);
+  // console.log('xScale', this._xScale.domain(), this.maxZoom);
+  // console.log('zoomLevel:', zoomLevel, this.tilesetInfo.min_pos[0],
+  //   this.tilesetInfo.max_pos[0]);
+
+  return zoomLevel;
+};
+
+
+/**
+ * Calculate the tiles that should be visible get a data domain
+ * and a tileset info
+ *
+ * All the parameters except the first should be present in the
+ * tileset_info returned by the server.
+ *
+ * @param zoomLevel: The zoom level at which to find the tiles (can be
+ *   calculated using this.calculateZoomLevel, but needs to synchronized across
+ *   both x and y scales so should be calculated externally)
+ * @param scale: A d3 scale mapping data domain to visible values
+ * @param minX: The minimum possible value in the dataset
+ * @param maxX: The maximum possible value in the dataset
+ * @param maxZoom: The maximum zoom value in this dataset
+ * @param maxDim: The largest dimension of the tileset (e.g., width or height)
+ *   (roughlty equal to 2 ** maxZoom * tileSize * tileResolution)
+ */
+const calculateTiles = (
+  zoomLevel, scale, minX, maxX, maxZoom, maxDim
+) => {
+  const zoomLevelFinal = Math.min(zoomLevel, maxZoom);
+
+  // the ski areas are positioned according to their
+  // cumulative widths, which means the tiles need to also
+  // be calculated according to cumulative width
+
+  const tileWidth = maxDim / (2 ** zoomLevelFinal);
+  // console.log('maxDim:', maxDim);
+
+  const epsilon = 0.0000001;
+
+  /*
+  console.log('minX:', minX, 'zoomLevel:', zoomLevel);
+  console.log('domain:', scale.domain(), scale.domain()[0] - minX,
+  ((scale.domain()[0] - minX) / tileWidth))
+  */
+
+  return range(
+    Math.max(0, Math.floor((scale.domain()[0] - minX) / tileWidth)),
+    Math.min(
+      2 ** zoomLevelFinal,
+      Math.ceil(((scale.domain()[1] - minX) - epsilon) / tileWidth),
+    ),
+  );
+};
+
+/**
+ * Calculate the tiles that sould be visisble given the resolution and
+ * the minX and maxX values for the region
+ *
+ * @param resolution: The number of base pairs per bin
+ * @param scale: The scale to use to calculate the currently visible tiles
+ * @param minX: The minimum x position of the tileset
+ * @param maxX: The maximum x position of the tileset
+ */
+const calculateTilesFromResolution = (resolution, scale, minX, maxX, pixelsPerTile) => {
```

Comment:
```suggestion
const calculateTilesFromResolution = (
  resolution, scale, minX, maxX = Number.MAX_VALUE, pixelsPerTile = 256
) => {
```

Some editors pick up default values and it's close to Python :)